### PR TITLE
add note about requiring markdown-it

### PR DESCRIPTION
**Pull request by arve0**
I've stumbled upon several plugins ([toc and anchor](https://github.com/MoOx/markdown-it-toc-and-anchor/blob/master/src/index.js#L1), [named headers](https://github.com/leff/markdown-it-named-headers/blob/master/index.js#L4), [anchor](https://github.com/valeriangalliat/markdown-it-anchor/blob/master/index.es6.js#L3)) which require markdown-it directly. In this case, it seems like markdown-it-anchor was the first to require markdown-it, then the others have followed. markdown-it-named-headers doesn't even use `Token`.

Tldr; I think this note is useful for fresh plugin writers.

### CODE REVIEW

**Comment by puzrin** [comment]
Those plugins require one class from markdown-it. It's not nesessary, but technically possible. I don't think such note will be clear.

**Comment by puzrin** [comment]
I think, you mean, plugins should not use markdown-it as dependency in config. Probably, it's better to write plugin authors, that they should use `token = md.inline.State.push(...)` instead of direct Token require. https://github.com/markdown-it/markdown-it/blob/master/lib/parser_block.js#L122

**Comment by arve0** [comment]
Ok, I can adjust the PR to your suggestions. Is `development.md` the correct place for the note?

**Comment by puzrin** [comment]
https://github.com/markdown-it/markdown-it/commit/b36c22600aee585a2685cac979f224b5e89df900

I've added rephrased note as stub. Feel free to change if needed.

**Comment by puzrin** [comment]
Thanks for your suggestion! May be this note will help to someone.
